underfs/cephfs/src/main/java/alluxio/underfs/cephfs/CephFSUnderFileSystem.java

```diff
@@ -0,0 +1,789 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.cephfs;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.ExceptionMessage;
+import alluxio.exception.InvalidPathException;
+import alluxio.retry.CountingRetry;
+import alluxio.retry.RetryPolicy;
+import alluxio.underfs.AtomicFileOutputStream;
+import alluxio.underfs.AtomicFileOutputStreamCallback;
+import alluxio.underfs.BaseUnderFileSystem;
+import alluxio.underfs.UfsDirectoryStatus;
+import alluxio.underfs.UfsFileStatus;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.options.CreateOptions;
+import alluxio.underfs.options.DeleteOptions;
+import alluxio.underfs.options.FileLocationOptions;
+import alluxio.underfs.options.MkdirsOptions;
+import alluxio.underfs.options.OpenOptions;
+import alluxio.util.UnderFileSystemUtils;
+import alluxio.util.io.PathUtils;
+
+import com.ceph.fs.CephFileAlreadyExistsException;
+import com.ceph.fs.CephMount;
+import com.ceph.fs.CephNotDirectoryException;
+import com.ceph.fs.CephStat;
+import com.ceph.fs.CephStatVFS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Stack;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * CephFS {@link UnderFileSystem} implementation.
+ */
+@ThreadSafe
+public class CephFSUnderFileSystem extends BaseUnderFileSystem
```

Comment:
Cephfs looks like a consistent under file system. For all consistent under file system that doesn't have the eventual consistency issue, we directly extends ConsistentUnderFileSystem. so that no need to duplicated override methods
```
  @Override
  public OutputStream createNonexistingFile(String path) throws IOException {
    return create(path);
  }

  @Override
  public OutputStream createNonexistingFile(String path, CreateOptions options) throws IOException {
    return create(path, options);
  }
```

Reply:
S3 is a storage that has eventual consistency issues, write - then - read may failed.